src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -189,4 +190,58 @@ public static Deadline parseDeadline(String deadline) throws ParseException {
         }
         return new Deadline(trimmedDeadline);
     }
+
+    /**
+     * Returns the day given in a string representing a date
+     */
+    public static int getDay(String date) {
+        // defensive programming
+        Matcher datetimeMatcher = Deadline.PATTERN_DATE.matcher(date);
+
+        if (datetimeMatcher.find()) {
+            return Integer.parseInt(datetimeMatcher.group(1));
+        } else {
+            return 0; // will eventually throw an error
```

Comment:
Year 0 is not accepted as valid, and if the user does input 0 it will be changed to year 2000 (by adding 2000); hence if 0 is returned the year was invalid and so `isValidDeadline` returns `true`.

This is just my reasoning, can you think of something that may cause an issue later?